eng/pipelines/common/global-build-job.yml

```diff
@@ -35,5 +36,14 @@ jobs:
     - template: /eng/pipelines/common/clone-checkout-bundle-step.yml
 
     # Build
-    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -arch ${{ parameters.archType }}
+    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }} -bl -ci
       displayName: Build product
+
+    - task: PublishBuildArtifacts@1
```

Comment:
Instead of having to add this step, you can just pass down a parameter to the template we use which is `job.yml` from arcade. That parameter is `enablePublishBuildArtifacts: true`. 

https://github.com/dotnet/arcade/blob/master/eng/common/templates/job/job.yml#L180

Note that you also need to declare a variable `_BuildConfig: ${{ parameters.buildConfig }}` because it uses it to find the logs:
https://github.com/dotnet/arcade/blob/master/eng/common/templates/job/job.yml#L184

Reply:
This job though deliberately builds multiple configurations though hence seems like it can't directly hook up to that infrastructure

Reply:
It can hook up to that infrastructure, the binlog will be created under the global buildConfig which is passed through the -configuration argument. The binlog name and location is calculated on the build script in arcade, which doesn't know about runtimeConfiguration. However if you want to produce binlogs for other steps in the coreclr build scripts, then we will definitely not be able to use this.